src/test/datascience/notebook/executionHelper.ts

```diff
@@ -0,0 +1,203 @@
+    public cells: TestNotebookCell[] = [];
+    public get cellCount(): number {
+        return this.cells.length;
+    }
+    constructor(
+        public readonly uri: Uri = Uri.file('untitled.ipynb'),
+        public readonly notebookType: typeof JupyterNotebookView | typeof InteractiveWindowView = JupyterNotebookView,
+        public metadata = { custom: {} },
+        public isUntitled = true,
+        public version: number = 0,
+        public isDirty = false,
+        public isClosed = false
+    ) {
+        MockNotebookDocuments.push(this);
+    }
+    static async openFile(uri: Uri): Promise<TestNotebookDocument> {
+        const notebook = await workspace.openNotebookDocument(uri);
+        // Its simpler to use VSCode to de-serialize an ipynb, else we need to write more code in the tests.
+        // This could be made faster by reading & parsing the ipynb ourselves instead of relying on VS Code.
+        const nb = new TestNotebookDocument(uri, JupyterNotebookView, notebook.metadata as any, false);
+        await Promise.all(
+            notebook.getCells().map((cell) => {
+                if (cell.kind === NotebookCellKind.Code) {
+                    return nb.appendCodeCell(cell.document.getText(), cell.document.languageId, cell.metadata);
+                } else {
+                    return nb.appendMarkdownCell(cell.document.getText(), cell.metadata);
+                }
+            })
+        );
+        return nb;
+    }
+    public async appendCodeCell(
+        content: string,
+        language: string = PYTHON_LANGUAGE,
+        metadata: { readonly [key: string]: any } = {}
+    ): Promise<TestNotebookCell> {
+        const textDoc = await workspace.openTextDocument({ language, content });
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        const cell = new TestNotebookCell(this, textDoc, NotebookCellKind.Code, 'text/plain', metadata);
+        this.cells.push(cell);
+        return cell;
+    }
+    public async insertCodeCell(
+        index: number,
+        content: string,
+        language: string = PYTHON_LANGUAGE,
+        metadata: { readonly [key: string]: any } = {}
+    ): Promise<TestNotebookCell> {
+        const textDoc = await workspace.openTextDocument({ language, content });
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        const cell = new TestNotebookCell(this, textDoc, NotebookCellKind.Code, 'text/plain', metadata);
+        this.cells.splice(index, 0, cell);
+        return cell;
+    }
+    public async appendMarkdownCell(
+        content: string,
+        metadata: { readonly [key: string]: any } = {}
+    ): Promise<TestNotebookCell> {
+        const textDoc = await workspace.openTextDocument({ language: 'markdown', content });
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        const cell = new TestNotebookCell(this, textDoc, NotebookCellKind.Markup, 'text/markdown', metadata);
```

Comment:
Not using VS Code edit APIs, that's a little slow and unnecessary
We're not trying to test VS Code here, we're testing Jupyter outputs, hence no need to use VS Code API